### DISCUSSION TITLE
Open webui always restarting afer install a reranking model

### DISCUSSION THREAD

**Discussion by Petracus** [open-webui/open-webui#5310] [General]
After trying to install a reranking model, I get this error when starting open webui:

File "/usr/local/lib/python3.11/site-packages/transformers/dynamic_module_utils.py", line 640, in resolve_trust_remote_code
    raise ValueError(
ValueError: Loading jinaai/jina-reranker-v2-base-multilingual requires you to execute the configuration file in that repo on your local machine. Make sure you have read the code there to avoid malicious use, then set the option `trust_remote_code=True` to remove this error.

Please, how can I remove that reranking model or set trust remote code=True?

Regards.

**Comment by silentoplayz** [comment]
https://docs.openwebui.com/getting-started/env-configuration#rag_embedding_model_trust_remote_code
https://docs.openwebui.com/getting-started/env-configuration#rag_reranking_model_trust_remote_code

**Comment by Petracus** [comment]
Yesterday I tried with RAG_RERANKING_MODEL_TRUST_REMOTE_CODE=true but no luck. 

I'll try again. 

Thanks.

**Comment by Donkey545** [comment]
For those facing the same issue with an incorrectly configured embedding or reranking model, you can change the value in the webui.db using sqlite3:

``` bash
#Switch to root so you can access the docker volumes
sudo su root

# Go to your database in in the volume:
cd /var/lib/docker/volumes/open-webui/_data

#Launch sqlite3 (install with: apt install sqlite3)
sqlite3

# perform the sqlite commands:
.open webui.db

#check which models are being used
SELECT json_extract(data, '$.rag.embedding_model') FROM your_table WHERE id = 1;
SELECT json_extract(data, '$.rag.reranking_model') FROM your_table WHERE id = 1;

#change the embedding model
UPDATE config
SET data = json_set(data, '$.rag.embedding_model', 'sentence-transformers/all-MiniLM-L6-v2')
WHERE id = 1;


#change the reranking model
UPDATE config
SET data = json_set(data, '$.rag.reranking_model', 'BAAI/bge-reranker-v2-m3')
WHERE id = 1;

```

This allows you to manually change that value when the container cannot launch due to a bad configuration as long as you mounted your _data into a volume when you originally launched the container. 

Remember that doing things as root is sketchy and can mess up permissions, so `su` back to your user when done or close the session.

**Reply by ovizii** [reply]
I would like to add to this:

> #check which models are being used
> SELECT json_extract(data, '$.rag.embedding_model') FROM your_table WHERE id = 1;
> SELECT json_extract(data, '$.rag.reranking_model') FROM your_table WHERE id = 1;

that `your_table` should be replaced with `config`.

**Comment by ovizii** [comment]
I was also seeing this error even though I had already set the value. 
> trust_remote_code=True


Anyway, even though the error displays, things work out. Kinda wonky but they work. 
I use an autoheal container which keeps restarting containers with degraded health. 

What happens is basically that open-webui becomes unhealthy while the download of the embedding model or reranking model is ongoing and autoheal restarts it, so the download keeps going. Depending on my internet speed, this takes like ^10-20 restarts of the container and then everything is back to perfectly working.
